src/components/button/button_content.tsx

```diff
@@ -109,9 +111,7 @@ export const OuiButtonContent: FunctionComponent<
 
   return (
     <span {...contentProps} className={contentClassNames}>
-      <span {...textProps} className={undefined}>
-        {buttonIcon}
-      </span>
+      {buttonIcon}
       <span {...textProps}>{children}</span>
```

Comment:
As mentioned before, no the wrapping `<span>` element is removed from `buttonIcon` element